src/WebJobs.Script/Environment/SystemEnvironment.cs

```diff
@@ -2,14 +2,14 @@
 // Licensed under the MIT License. See License.txt in the project root for license information.
 
 using System;
-using System.Collections.Generic;
-using System.Text;
+using System.Collections.Concurrent;
 
 namespace Microsoft.Azure.WebJobs.Script
 {
     public class SystemEnvironment : IEnvironment
     {
         private static readonly Lazy<SystemEnvironment> _instance = new Lazy<SystemEnvironment>(CreateInstance);
+        private static ConcurrentDictionary<string, string> _cache = new ConcurrentDictionary<string, string>(StringComparer.OrdinalIgnoreCase);
```

Comment:
A long time ago, we moved to our own environment impl because the default ASP.NET impl caches, and that doesn't work for us in light of specialization. We can add back caching to improve perf, as long as we ensure we reset cache on specialization as I'm now doing.